dlt/pipeline/pipeline.py

```diff
@@ -172,6 +173,40 @@ def _wrap(self: "Pipeline", *args: Any, **kwargs: Any) -> Any:
     return _wrap  # type: ignore
 
 
+def with_plugins() -> Callable[[TFun], TFun]:
+    def decorator(f: TFun) -> TFun:
+        @wraps(f)
+        def _wrap(self: "Pipeline", *args: Any, **kwargs: Any) -> Any:
+            is_new_context = self._plugin_ctx is None
+            if self._plugin_ctx is None:
+                self._plugin_ctx = PluginsContext()
+                self._plugin_ctx.setup_plugins(self.plugins)
+                self._plugin_ctx.on_start(self)
```

Comment:
what is step and how `on_step_start` differs from `on_start` should it be something like `on_init`?